README.md
# Dart SDK

## Availability
Supports Dart from 2.0 to above

## How to install

Add the following line in dependencies in your pubspec.yaml file and save it:
```
dependencies:
  nps_sdk: "^1.0.0"
```

If ```pub get``` wasn't automatically ejecuted, you can do it manually by opening a terminal in your project and executing the command:
```
pub get
```

Now you are ready to import our SDK:
```dart
import 'package:nps_sdk/nps_sdk.dart';
```

## Configuration

It's a basic configuration of the SDK

```dart
import 'dart:async';

import 'package:nps_sdk/nps_sdk.dart';

main(List<String> arguments) async {
  Nps nps = new Nps(sandbox);
}
```

Here is an simple example request:
```dart
import 'dart:async';

import 'package:nps_sdk/nps_sdk.dart';

main(List<String> arguments) async {
  Nps nps = new Nps(sandbox);

  Map createPaymentMethodTokenParams = {
    "psp_Version": "2.2",
    "psp_MerchantId": "sdk_test",
    "psp_CardInputDetails": {
      "Number": "4507990000000010",
      "ExpirationDate": "2501",
      "SecurityCode": "123",
      "HolderName": "JOHN DOE"
    },
    "psp_ClientSession": "YOUR_CLIENT_SESSION_HERE"
  };

  Future response = nps.createPaymentMethodToken(nps, createPaymentMethodTokenParams);
  response.then((resp) {
            print(resp["psp_ResponseCod"]); //Only one key
            jsonPrettyPrint(resp); //All the XML
          })
          .catchError((error) => print);
}
```


## Environments

```dart
import 'package:nps_sdk/nps_sdk.dart';

Nps nps = new Nps(sandbox);
Nps nps = new Nps(staging);
Nps nps = new Nps(production);
```

## Error handling

With this SDK you will be able to handle errors as you prefer.
In order to do this, you just need to specify what you want to do with the error in the catchError callback:
```dart
import 'dart:async';

import 'package:nps_sdk/nps_sdk.dart';

main(List<String> arguments) async {
    //Code
    
    Future response = nps.createPaymentMethodToken(nps, createPaymentMethodTokenParams);
      response.then((resp) => print(resp["psp_ResponseCod"]))
              .catchError((error) {
                //Your code to handle the error
              });
}
```

## Flutter 

This SDK can be imported in Flutter's projects. For a Flutter example, click [here](https://github.com/Ingenico-NPS-Latam/nps-sdk-dart/tree/master/example/flutter_example).



CHANGELOG.md
## 1.0.0
- Upgrade to Dart 2

## 0.0.9
- Fixed pub warnings 

## 0.0.8
- Added Flutter example 

## 0.0.7
- Changed xml's library version

## 0.0.6
- Refactored code

## 0.0.5
- Added methods for validation
- Changed hosts' paths
- Fixed getInstallmentsOptions method

## 0.0.4
- Made changes in README.md

## 0.0.3
- Made changes in README.md 

## 0.0.2
- Made changes in the example

## 0.0.1
- First SDK's commit

example/flutter_example/lib/home_page.dart
import 'package:nps_sdk/nps_sdk.dart';

class HomePage extends StatefulWidget {
  HomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _HomePageState createState() => new _HomePageState();
}

class _HomePageState extends State<HomePage> {
  //Text Controllers
  TextEditingController _nameController;
  TextEditingController _cardController;
  TextEditingController _cvvController;
  TextEditingController _expController;
  
  //Text Variables 
  String _introText = "In this demo the user has chosen to pay with credit card so the payment form has been displayed";
  String _name = "Jhon Smith";
  String _card = "4507990000000010";
  String _cvv = "123";
  String _expDate = "1912";
  String _responseMsg = "";

  @override
  void initState() {
    //Carga la infomacion cuando se abre la aplicacion
    super.initState();
    _nameController = new TextEditingController(text: _name);
    _cardController = new TextEditingController(text: _card);
    _cvvController = new TextEditingController(text: _cvv);
    _expController = new TextEditingController(text: _expDate);
  }

  _createPaymentMethodToken() async {
    Nps nps = new Nps(sandbox);

    Map createPaymentMethodTokenParams = {
      "psp_Version": "2.2",
      "psp_MerchantId": "sdk_test",
      "psp_CardInputDetails": {
        "HolderName": _nameController.text,
        "Number": _cardController.text,
        "SecurityCode": _cvvController.text,
        "ExpirationDate": _expController.text
      },
      "psp_ClientSession": "YOUR_CLIENT_SESSION"
    }; 

    var response = await nps.createPaymentMethodToken(nps, createPaymentMethodTokenParams);

    setState(() {
      _responseMsg = response["psp_ResponseMsg"];
    });
  }

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text(widget.title),
      ),
      body: new Container(
        child: ListView(
          children: <Widget>[
            new Text(_introText),
            new Container(
              margin: new EdgeInsets.only(left: 120.0, right: 120.0, top: 5.0,),
              child:  new TextField(
                controller: _nameController,
              ),
            ),
            new Container(
              margin: new EdgeInsets.only(left: 120.0, right: 120.0),
              child:  new TextField(
                controller: _cardController,
              ),
            ),
            new Container(
              margin: new EdgeInsets.only(left: 120.0, right: 120.0),
              child:  new TextField(
                controller: _cvvController,
              ),
            ),
            new Container(
              margin: new EdgeInsets.only(left: 120.0, right: 120.0),
              child:  new TextField(
                controller: _expController,
              ),
            ),
            new Container(
              margin: new EdgeInsets.all(15.0),
                child: new Text(_responseMsg, textAlign: TextAlign.center, style: TextStyle(fontSize: 18.0, fontWeight: FontWeight.bold),), 
            ),
            new Container(
              margin: new EdgeInsets.only(left: 150.0, right: 150.0),
              child: new RaisedButton(
                child: new Text("SEND"),
                onPressed: _createPaymentMethodToken
              ), 
            )
          ],
        ),
      ),
    );
  }
}

